Classes/Utility/UriUtility.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace ApacheSolrForTypo3\Solr\Utility;
+
+/*
+ * This file is part of the TYPO3 CMS project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+/**
+ * This utility class contains several methods for URI handling
+ *
+ * @author Lars Tode <lars.tode@dkd.de>
+ */
+class UriUtility
+{
+    /**
+     * Converts a query string into an array.
+     *
+     * @param string $query
+     * @return array
+     */
+    public static function queryStringToArray(string $query): array
+    {
+        if ($query === '') {
+            return [];
+        }
+
+        return explode('&', $query);
+    }
+    /**
+     * Converts a query array into a string.
+     *
+     * @param array $query
+     * @return string
+     */
+    public static function queryArrayToString(array $query): string
+    {
+        return implode('&', $query);
```

Comment:
check out https://www.php.net/manual/en/function.http-build-query.php this is much more graceful with errors and multi-dimensional logic, and also adheres to RFC standards (see the fourth argument there, and how TYPO3 Core uses it in HttpUtility)

Reply:
The static method dropped in favot for the PHP build-in function.